src/sentry/search/events/fields.py

```diff
@@ -464,8 +463,9 @@ def resolve_field_list(
                         if function.details.instance.redundant_grouping:
                             aggregate_fields[format_column_as_key(function.aggregate[1])].add(field)
 
+    check_aggregations = snuba_filter.having and len(aggregations) > 0
```

Comment:
nit: if we checked `snuba_filter.condition_aggregates` here as well, we can avoid the `set(... or [])` pattern on L468 entirely